### PR TITLE
added account history hook

### PR DESCRIPTION
**Pull request by poocart**
https://linear.app/pillarproject/issue/PRO-1384/transactionkit-transactions-history-hook

Adds `useEtherspotHistory` to fetch Etherspot Account transactions.

### CODE REVIEW

File: __tests__/hooks/useEtherspotHistory.test.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+// hooks
+import { useEtherspotHistory, EtherspotUi } from '../../src';
+
+describe('useEtherspotHistory()', () => {
+  it('getAccountTransactions() returns account history', async () => {
+    const wrapper = ({ children }) => (
+      <EtherspotUi provider={null}>
+        {children}
+      </EtherspotUi>
```
Comment (IAmKio) [inline __tests__/hooks/useEtherspotHistory.test.js:11]: Should this be `EtherspotTransactionKit`?

File: __tests__/hooks/useEtherspotHistory.test.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    rerender({ chainId: 137 });
+
+    const accountTransactionsPolygon = await result.current.getAccountTransactions();
+    expect(accountTransactionsPolygon.length).toEqual(0);
+  });
+  it('getAccountTransaction() returns transaction by existing hash', async () => {
+    const wrapper = ({ children }) => (
+      <EtherspotUi provider={null}>
+        {children}
+      </EtherspotUi>
```
Comment (IAmKio) [inline __tests__/hooks/useEtherspotHistory.test.js:38]: As above

File: src/hooks/useEtherspotHistory.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    }
+
+    let transactions: Transaction[] = [];
+
+    try {
+      ({ items: transactions } = await sdkForChainId.getTransactions({
+        account: sdkForChainId.state.account.address,
+      }));
+    } catch (e) {
+      console.warn(e);
```
Comment (IAmKio) [inline src/hooks/useEtherspotHistory.ts:32]:
```suggestion
      console.warn(`Sorry, an error occurred whilst trying to fetch transactions for ${sdkForChainId.state.account.address}. Please try again. Error:`, e);
```

File: src/hooks/useEtherspotHistory.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    if (!sdkForChainId) return;
+
+    if (sdkForChainId?.state?.account?.type !== AccountTypes.Contract) {
+      await connect(chainId);
+    }
+
+    try {
+      return sdkForChainId.getTransaction({ hash });
+    } catch (e) {
+      console.warn(e);
```
Comment (IAmKio) [inline src/hooks/useEtherspotHistory.ts:49]:
```suggestion
console.warn(`Sorry, an error occurred whilst trying to fetch the transaction for ${sdkForChainId.state.account.address}. Please try again. Error:`, e);
```

**Review by IAmKio — CHANGES_REQUESTED** [review]
Just a few small things but other than that - looks great!

File: __tests__/hooks/useEtherspotHistory.test.js (hunk 1 above)
Comment (poocart) [reply __tests__/hooks/useEtherspotHistory.test.js]: Oops sorry, was not updated to master, fixed!

File: __tests__/hooks/useEtherspotHistory.test.js (hunk 2 above)
Comment (poocart) [reply __tests__/hooks/useEtherspotHistory.test.js]: Oops sorry, was not updated to master, fixed!

**Review by IAmKio — APPROVED** [review]
👍 🍔

**Pull request merged**